src/store/useManualRunStore.ts

```diff
@@ -27,48 +33,30 @@ const store: StateCreator<ManualRunStore> = (set, get) => ({
 		{ entrypointFunction, filePath, files, isJson, isManualRunEnabled, lastDeployment, params }
 	) => {
 		set((state) => {
-			const projectData = state.projectManualRun[projectId] || { ...defaultProjectState };
-
-			if (isManualRunEnabled !== undefined) {
-				projectData.isManualRunEnabled = isManualRunEnabled;
-			}
+			const projectData = {
+				...createDefaultProjectState(),
+				...state.projectManualRun[projectId],
+			};
 
 			if (files) {
-				projectData.files = files;
-
-				const fileOptions = files.map((file) => ({
-					label: file,
-					value: file,
-				}));
-
-				projectData.fileOptions = fileOptions;
-
-				const firstFile = fileOptions[0];
-				projectData.filePath = firstFile;
-				projectData.entrypointFunction = "";
-				projectData.params = [];
-			}
-
-			if (filePath) {
-				projectData.filePath = filePath;
-				projectData.entrypointFunction = "";
+				const fileOptions = createFileOptions(files);
```

Comment:
I'm not sure we need to create a new function for this, `map` is only used in one space

Reply:
done